src/xdpinstaller/Product.wxs

```diff
@@ -0,0 +1,91 @@
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:ui="http://schemas.microsoft.com/wix/UIExtension" xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
+	<Product Id="$(var.ProductVersion)" Name="XDP for Windows" Language="1033" Version="0.16.9" Manufacturer="Microsoft" UpgradeCode="$(var.UpgradeCode)">
+		<Package Description="XDP for Windows" InstallerVersion="301" Compressed="yes" InstallScope="perMachine" Manufacturer="Microsoft" Platform="x64" />
+		<MajorUpgrade AllowSameVersionUpgrades="yes"
+					  Disallow="yes" DisallowUpgradeErrorMessage="An older version of [ProductName] is already installed. Please remove it first."
+					  AllowDowngrades="no" DowngradeErrorMessage="A newer version of [ProductName] is already installed. Please remove it first." Schedule="afterInstallFinalize" />
+		<MediaTemplate EmbedCab="yes" />
+
+		<!-- Define global properties -->
+		<PropertyRef Id="WIX_ACCOUNT_LOCALSERVICE" />
+		<Property Id="ARPCONTACT" Value="opencode@microsoft.com" />
+		<Property Id="TARGETDIR" Secure="yes">
+			<RegistrySearch Id="FindInstallLocation" Root="HKLM" Key="Software\Microsoft\Windows\CurrentVersion\Uninstall\[WIX_UPGRADE_DETECTED]" Name="InstallLocation" Type="raw" Win64="yes" />
+		</Property>
+		
+		<!-- Define the Product features and installation steps -->
+		<Feature Id="ProductFeature" Title="XDP for Windows Installer" ConfigurableDirectory="TARGETDIR" Display="expand" Level="1" Absent="disallow" AllowAdvertise="no" InstallDefault="local" TypicalDefault="install" >	
+			<ComponentGroupRef Id="xdp_driver" />
+			<ComponentGroupRef Id="xdpapi_dll" />
+		</Feature>
+		
+		<InstallExecuteSequence>
+			<!--Rollback sequence-->
+			<Custom Action="xdp_uninstall_rollback" Before="xdp_install">NOT Installed</Custom>
+
+			<!--Install sequence-->
+			<Custom Action="xdp_install" After="InstallFiles">NOT Installed</Custom>
+
+			<!--Uninstall sequence-->
+			<Custom Action="xdp_uninstall" After="InstallInitialize">REMOVE="ALL"</Custom>
+		</InstallExecuteSequence>
+
+		<!-- Define the UI style & behavior -->
+		<UIRef Id="WixUI_Minimal" />
+		<WixVariable Id="WixUILicenseRtf" Value="$(var.SolutionDir)..\..\LICENSE.rtf" />
+		<InstallUISequence/>
+		<UI/>
+	</Product>
+
+	<!-- Define installation directories -->
+	<Fragment>
+		<Directory Id="TARGETDIR" Name="SourceDir">
+			<Directory Id="System64Folder" Name="System64Folder">
+				<Directory Id="dir_drivers" Name="drivers"/>
+			</Directory>
+		</Directory>
+	</Fragment>
+
+	<!-- Define the product components -->
+	<Fragment>
+		<!-- XDP driver components -->
+		<ComponentGroup Id="xdp_driver" Directory="dir_drivers">
+			<Component Id="xdp_setup" Guid="{0D991B56-5C01-4E57-AB33-E81696B73D55}">
+				<File Id="xdp_setup_ps1" Source="scripts\xdp-setup.ps1" KeyPath="yes"/>
+			</Component>
+			<Component Id="xdp.inf" Guid="{70A53BC2-8D56-4D53-B8FA-F07EC3D4783A}">
+				<File Id="xdp.inf" Name="xdp.inf" Source="$(var.TargetDir)xdp.inf" KeyPath="yes"/>
```

Comment:
Is this copying the .inf file into system32\drivers? This shouldn't be required - the INF can be installed from anywhere. Same with the .cat file.

Reply:
The MSI embeds all the files that need to be deployed: there would be no file to operate on if the Windows installer does not deploy what's packaged.
Also, that's why there's a modified INF file, which skips the deployment so that netcfg will not fail to install the driver.

Reply:
Let's copy all files to c:\program files\xdp rather than subdirectories of c:\windows\system32. We can install the INF and the service directly from c:\program files, and we should register xdpapi.dll so it can be automatically loaded by other programs.

Reply:
Yep, as discussed internally, that's the right direction (and the one taken for eBPF for Windows).
I'll add a system wide path to the install dir, and register the DLL.

Reply:
Unfortunately, there seems to be an issue with the xdp driver installation, when done from folders out of `%SystemRoot%\drivers`:

```
PS C:\_xdp\_tmp> dir

    Directory: C:\_xdp\_tmp

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         6/21/2023   2:30 PM           4045 xdp-setup.ps1
-a----         6/21/2023   2:25 PM           1932 xdp.cat
-a----         6/21/2023   2:25 PM           4084 xdp.inf
-a----         6/21/2023   2:25 PM         423840 xdp.sys

PS C:\_xdp\_tmp> .\xdp-setup.ps1 -Install xdp -Verbose
VERBOSE: netcfg.exe -v -l 'C:\_xdp\_tmp\xdp.inf' -c s -i ms_xdp
VERBOSE: Trying to install ms_xdp ...
VERBOSE:
VERBOSE:
VERBOSE: ... C:\_xdp\_tmp\xdp.inf was copied to C:\Windows\INF\oem1.inf.
VERBOSE:
VERBOSE:
VERBOSE: ... failed. Error code: 0x800f0217.
VERBOSE:
VERBOSE:
C:\_xdp\_tmp\xdp-setup.ps1 : netcfg.exe exit code: 535
At line:1 char:1
+ .\xdp-setup.ps1 -Install xdp -Verbose
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,xdp-setup.ps1
```

Therefore, the current latest commit implements a workaround that will still deploy in `%ProgramFiles%\xdp` (which is now also cinfigurable!), *except* for `xdp.sys`, which will be deployed in `%SystemRoot%\drivers`.

Once the investigation on the driver installation issue is found, then just changing the deploy folder for `xdp.sys` to `%ProgramFiles%\xdp\drivers`  (within `Product.wxs`) would be required.

Reply:
Awesome, thanks!!!